cheatsheets/gleam-for-python-users.md

```diff
@@ -0,0 +1,1052 @@
+
+An exception is raised using the keyword `raise`.
+
+```py
+def an_fn_that_fails():
+    raise Exception('an error')
+```
+
+The callee block will be able to capture any exception raised in the block using a `try/except` set of blocks:
+
+```py
+try:
+    print('executed')
+    an_fn_that_fails()
+    print('not_executed')
+except Exception as e:
+    print('doing something with the exception', e)
+
+```
+
+#### Gleam
+
+In contrast in gleam, errors are just containers with an associated value.
+
+A common container to model an operation result is `Result(ReturnType, ErrorType)`.
+
+A Result is either:
+
+- an `Error(ErrorValue)`
+- or an `Ok(Data)` record
+
+Handling errors actually means to match the return value against those two scenarios, using a case for instance:
+
+```rust
+case parse_int("123") {
+  Error(e) -> io.println("That wasn't an Int")
+  Ok(i) -> io.println("We parsed the Int")
+}
+```
+
+In order to simplify this construct, we can use the `try` keyword that will:
+
+- bind a value to the providing name if Ok(Something) is matched
+- **interrupt the flow** and return `Error(Something)`
+
+```rust
+let a_number = "1"
+let an_error = Error("ouch")
+let another_number = "3"
+
+try int_a_number = parse_int(a_number)
+try attempt_int = parse_int(an_error) // Error will be returned
+try int_another_number = parse_int(another_number) // never gets executed
+
+Ok(int_a_number + attempt_int + int_another_number) // never get executed
+```
+
+## Type aliases
+
+Type aliases allow for easy referencing of arbitrary complex types. Even though their type systems does not serve the same function, both Python and Gleam provide this feature.
+
+### Python
+
+A simple variable can store the result of a compound set of types.
+
+```py
+import List from typing
+
+Vector = List[float]
+
+# can now be used to annotate a variable
+tensor: Vector = [1.0, 2.0, 3.0]
+```
+
+### Gleam
+
+The `type` keyword can be used to create aliases
+
+```rust
+pub type Headers =
+  List(tuple(String, String))
+```
+
+## Custom types
+
+### Records
+
+Custom type allows you to define a collection data type with a fixed number of named fields, and the values in those fields can be of differing types.
+
+#### Python
+
+Python uses classes to define user-defined, record-like types.
+Properties are defined as class members and initial values are generally set in the constructor.
+
+By default the constructor does not provide base initializers in the constructor so some boilerplate is needed:
+
+```python
+class Person():
+    name: str
+    age: int
+    def __init__(name: str, age: int):
+        self.name = name
+        self.age = int
+
+person = Person(name="Jake", age=20)
+# or with positional arguments Person("Jake", 20)
+name = person.name
+```
+
+A more recent alternative is to leverage the `NamedTuple` base type to generate a constructor with initializers.
```

Comment:
This is cool!

It would be good to note also that in Gleam there is no OOP, so methods cannot be added to types